src/meta-srv/src/procedure/region_failover.rs

```diff
@@ -234,6 +234,16 @@ impl RegionFailoverManager {
             })
             .unwrap_or_default())
     }
+    #[allow(dead_code)]
+    pub(crate) async fn is_maintenance_node(&self) -> Result<bool> {
+        let key = MAINTENANCE_KEY.to_string().into_bytes();
+
+        self.selector_ctx
+            .kv_store
```

Comment:
Better to check the maintenance mode in the `in_memory` store. Check from `kv_store` everytime is costly.